apps/session-manager/src/controllers/fast-login.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessJson,
+} from "@pagopa/ts-commons/lib/responses";
+import { flow, pipe } from "fp-ts/function";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as RTE from "fp-ts/ReaderTaskEither";
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
+import { GenerateNonceResponse } from "../generated/fast-login-api/GenerateNonceResponse";
+import { readableProblem } from "../utils/errors";
+import {
+  IResponseErrorUnauthorized,
+  ResponseErrorStatusNotDefinedInSpec,
+  ResponseErrorUnexpectedAuthProblem,
+} from "../utils/responses";
+import { getFastLoginLollipopConsumerClient } from "../repositories/fast-login-api";
+
+export const generateNonceEndpoint: RTE.ReaderTaskEither<
+  {
+    client: ReturnType<getFastLoginLollipopConsumerClient>;
+  },
+  Error,
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+  | IResponseSuccessJson<GenerateNonceResponse>
+> = ({ client }) =>
+  TE.tryCatch(
+    async () =>
+      pipe(
+        TE.tryCatch(
```

Comment:
the double tryCatch can be avoided if the left path of the inner TE is Error with a final map to `ResponseErrorInternal`, the other response can be mapped on the right path because no other action are executed after the final chain

Reply:
refactored in https://github.com/pagopa/io-auth-n-identity-domain/pull/18/commits/8a083e326d241baa8febfd77f182b4588be698bc